scripts/update_project_keywords.py

```diff
@@ -0,0 +1,93 @@
+"""Update the project keywords to include a list of supported device models."""
+
+import json
+import logging
+import os
+import pathlib
+import re
+import sys
+from typing import Any
+
+import click
+import tomlkit
+
+_logger = logging.getLogger(__name__)
+
+
+def _get_logging_level(verbose, quiet):
+    if 0 < verbose and 0 < quiet:
+        click.exceptions.error("Mixing --verbose and --quiet is contradictory")
+    verbosity = 2 + quiet - verbose
+    verbosity = max(verbosity, 0)
+    verbosity = min(verbosity, 4)
+    logging_level = {
+        0: logging.DEBUG,
+        1: logging.INFO,
+        2: logging.WARNING,
+        3: logging.ERROR,
+        4: logging.CRITICAL,
+    }[verbosity]
+
+    return logging_level
+
+
+@click.command()
+@click.option("--verbose", type=int, default=0, required=False)
+@click.option("--quiet", type=int, default=0, required=False)
+def main(verbose, quiet):
+    """Update the project keywords to include a list of supported device models."""
+    logging_level = _get_logging_level(verbose, quiet)
+
+    log_format = "[%(relativeCreated)6d] %(levelname)-5s %(funcName)s: %(message)s"
+    logging.basicConfig(level=logging_level, format=log_format)
+
+    if sys.platform != "win32":
+        raise click.ClickException(f"Unsupported platform: {sys.platform}")
+
+    device_caps = _read_device_caps()
+    supported_products = _get_supported_products(device_caps)
+    pyproject_path = pathlib.Path("pyproject.toml")
+    pyproject_toml = tomlkit.parse(pyproject_path.read_text(encoding="utf-8"))
+    keywords = pyproject_toml["tool"]["poetry"]["keywords"]
+    # Replace everything after "daq".
+    daq_index = keywords.index("daq")
+    new_keywords = keywords[: daq_index + 1] + sorted(supported_products)
+    pyproject_toml["tool"]["poetry"]["keywords"] = new_keywords
+    pyproject_path.write_text(tomlkit.dumps(pyproject_toml), encoding="utf-8")
+
+
+def _read_device_caps() -> dict[str, Any]:
+    program_files_path = pathlib.Path(os.environ["ProgramFiles"])
+    instrumentstudio_path = program_files_path / "National Instruments" / "InstrumentStudio"
```

Comment:
On further thought, it would make sense to add this file to the repo near the codegen metadata so this functionality can be added to the codegen script and checked in build.yml.